Add workaround for failing pip on Ubuntu/Debian

jan-schulz-k24 opened the pull request:
On a flaky network connections pip tries to restart a process and on
debian/ubuntu then fails with:

    TypeError: unsupported operand type(s) for -=: 'Retry' and 'int'

The original error is explained in
https://stackoverflow.com/a/46970344/1380673 and has something to do with un-
vendored requests/urllib3 on Debian and Ubuntu. Here we simply force pip to
use the newly installed requests/urllib3 instead of the one which came from
the host system.

Closes: #19

jacopofar commented:
Good!
(I think the test fails because the testing branch is not merged yet and Travis is using the default Ruby config)

jankatins committed:
Add workaround for failing pip on Ubuntu/Debian

On a flaky network connections pip tries to restart a process and on
debian/ubuntu then fails with:

    TypeError: unsupported operand type(s) for -=: 'Retry' and 'int'

The original error is explained in
https://stackoverflow.com/a/46970344/1380673 and has something to do with un-
vendored requests/urllib3 on Debian and Ubuntu. Here we simply force pip to
use the newly installed requests/urllib3 instead of the one which came from
the host system.

Closes: #19

ghost commented:
@martin-loetzsch I rebased the patch on top of current master. I would really liek to get this merged to be able to checkout on ubuntu without manually creating the venv...
